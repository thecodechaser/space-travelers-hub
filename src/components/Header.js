import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../asset/logo.png';

const Header = () => {
  const links = [
    {
      text: 'Rockets',
      path: '/',
      id: 0,
    },
    {
      text: 'Missions',
      path: '/missions',
      id: 1,
    },
    {
      text: 'My Profile',
      path: '/myprofile',
      id: 2,
    },
  ];
  return (
    <header className="header">
      <div className="logo-title-c">
        <img alt="logo-top" src={logo} className="logo-top" />
        <h3 className="header-title">Space Traveler&apos;s Hub</h3>
      </div>
      <nav>
        <ul className="nav-menu">
          {
          links.map((link) => (
            <li key={link.id}>
              <NavLink to={link.path} className="nav-item">{link.text}</NavLink>
            </li>
          ))
        }
        </ul>
      </nav>
    </header>
  );
};

export default Header;
