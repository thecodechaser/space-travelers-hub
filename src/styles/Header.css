/* style for header */

.header {
  padding: 0 40px;
  display: flex;
  justify-content: space-between;
  border-bottom: 2px solid rgb(209, 209, 209);
  margin: 0 30px;
}

.logo-title-c {
  display: flex;
  gap: 20px;
}

.logo-top {
  width: 80px;
  height: 80px;
}

.header-title {
  margin-top: 25px;
  font-size: 30px;
}

.nav-menu {
  list-style-type: none;
  display: flex;
  gap: 15px;
  margin-top: 25px;
}

.nav-item {
  text-decoration: none;
  font-size: 18px;
  color: #0290ff;
}

.nav-item:hover {
  color: #005ca3;
}

.nav-item:focus {
  text-decoration: underline;
  color: #005ca3;
}

.nav-menu li:last-child {
  border-left: 2px solid #000;
  padding-left: 15px;
}
