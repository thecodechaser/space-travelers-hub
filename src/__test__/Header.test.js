import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Header from '../components/Header';

describe('Header tests', () => {
  test('should render', () => {
    const header = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );

    expect(header).toMatchSnapshot();
  });
});
