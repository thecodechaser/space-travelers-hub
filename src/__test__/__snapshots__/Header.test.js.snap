// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="logo-title-c"
        >
          <img
            alt="logo-top"
            class="logo-top"
            src="logo.png"
          />
          <h3
            class="header-title"
          >
            Space Traveler's Hub
          </h3>
        </div>
        <nav>
          <ul
            class="nav-menu"
          >
            <li>
              <a
                aria-current="page"
                class="nav-item active"
                href="/"
              >
                Rockets
              </a>
            </li>
            <li>
              <a
                class="nav-item"
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li>
              <a
                class="nav-item"
                href="/myprofile"
              >
                My Profile
              </a>
            </li>
          </ul>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="logo-title-c"
      >
        <img
          alt="logo-top"
          class="logo-top"
          src="logo.png"
        />
        <h3
          class="header-title"
        >
          Space Traveler's Hub
        </h3>
      </div>
      <nav>
        <ul
          class="nav-menu"
        >
          <li>
            <a
              aria-current="page"
              class="nav-item active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class="nav-item"
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class="nav-item"
              href="/myprofile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
